contracts/erc20-gateway/ERC20Gateway.sol

```diff
@@ -80,4 +80,34 @@ contract ERC20Gateway is MasterCopyNonUpgradable, MessageBus, ERC20GatewayBase {
         );
     }
 
+    /**
+     * @notice This method verifies merkle proof of ERC20Cogateway contract.
+     *
+     * @param _blockNumber Block number at which ERC20Cogateway contract
+     *                     address is to be proven.
+     * @param _rlpAccount RLP encoded account node object.
+     * @param _rlpParentNodes RLP encoded value of account proof node array.
+     *
+     * \post Calls `MessageInbox.proveStorageAccount()` function with
+     *       `_blockNumber`, `_rlpAccountNode`, `_rlpParentNodes` as input
+     *       parameters.
+     * \post Emits `GatewayProven` event with parameters `address(this)` and
+     *       `_blockNumber`.
+     */
+    function proveGateway(
+        uint256 _blockNumber,
+        bytes calldata _rlpAccount,
+        bytes calldata _rlpParentNodes
+    )
+        external
+    {
+        MessageInbox.proveStorageAccount(
+            _blockNumber,
+            _rlpAccount,
+            _rlpParentNodes
+        );
+
+        emit GatewayProven(address(this), _blockNumber);
```

Comment:
This should emit the `Cogateway` address.

Reply:
Done!